### PR TITLE
Update dev mode

### PR DESCRIPTION
**Pull request by futuretea**

### CODE REVIEW

File: docs/dev/dev-mode.md [hunk 1]
```diff
@@ -13,70 +13,103 @@ Description: Developer mode (dev mode) is intended to be used for testing and de
...
 
+- The Kubernetes node must pass the [host-check](https://raw.githubusercontent.com/harvester/harvester/master/hack/host-check.sh)
+- If the Kubelet's RootDir is not `/var/lib/kubelet`, it must be mounted bind to `/var/lib/kubelet`
+   ```bash
+   KUBELET_ROOT_DIR="replace with your kubelet root dir"
+   echo "${KUBELET_ROOT_DIR} /var/lib/kubelet none bind 0 0" >> /etc/fstab
+   mkdir -p /var/lib/kubelet && mount -a
+   ```
+- The Kubernetes controller-manager args requires `--use-service-account-credentials=true`
+- To display the CPU, memory, and disk usage of the host properly, the [metrics-server](https://github.com/kubernetes-sigs/metrics-server) must be installed in the cluster.
```
Comment (nunix) [inline docs/dev/dev-mode.md:26]:
```suggestion
- The Kubernetes node must pass the [host-check](https://raw.githubusercontent.com/harvester/harvester/master/hack/host-check.sh)
- If the Kubelet's RootDir is not `/var/lib/kubelet`, you must create a `bind mount` to `/var/lib/kubelet` as follow:
   ```bash
   KUBELET_ROOT_DIR="path to your kubelet root dir"
   echo "${KUBELET_ROOT_DIR} /var/lib/kubelet none bind 0 0" >> /etc/fstab
   mkdir -p /var/lib/kubelet && mount -a
   ```
- The Kubernetes controller-manager args requires `--use-service-account-credentials=true`
- To display the CPU, memory, and disk usage of the host properly, the [metrics-server](https://github.com/kubernetes-sigs/metrics-server) must be installed in the cluster.
```

Comment (nunix) [reply docs/dev/dev-mode.md:26]: For the `Kubernetes controller-manager` point, can we add an example or a reference which explains how/where it should be set

**Review by nunix — COMMENTED** [review]
Rephrased to active voice and the `bind mount` point

File: docs/dev/dev-mode.md (hunk 1 above)
Comment (jtravee) [reply docs/dev/dev-mode.md:26]: Small nit, line 19: "as follows:"

File: docs/dev/dev-mode.md [hunk 2]
```diff
@@ -13,70 +13,103 @@ Description: Developer mode (dev mode) is intended to be used for testing and de
...
+   --set harvester-load-balancer.enabled=true \
+   --set kube-vip.enabled=true \
+   --set kube-vip.config.vip_interface=${VIP_NIC} \
+   --set kube-vip.config.vip_address=${VIP_IP} \
+   --set service.vip.enabled=true \
+   --set service.vip.ip=${VIP_IP}
+   ```
+
+   ```bash
+   ## In some kubernetes distributions (such as kubeadm), we need to modify the kube-vip nodeSelector to match the control-plane nodes
```
Comment (jtravee) [inline docs/dev/dev-mode.md:101]:
```suggestion
   ## In some Kubernetes distributions (such as kubeadm), we need to modify the kube-vip nodeSelector to match the control-plane nodes.
```

File: docs/dev/dev-mode.md [hunk 3]
```diff
@@ -13,70 +13,103 @@ Description: Developer mode (dev mode) is intended to be used for testing and de
...
-1. Click **Create**.
-1. Navigate to your project-level `Apps`.
-1. Click **Launch** and choose the Harvester app.
-1. (Optional) You can modify the configurations if needed. Otherwise, use the default options.
-1. Click **Launch** and wait for the app's components to be ready.
-1. Click the **/index.html** link to navigate to the Harvester UI, as shown in the figure below:
-   ![harvester-app.png](harvester-app.png)
+1. Expose Harvester UI
+   ```bash
+   ## refer to https://kube-vip.chipzoller.dev/docs/usage/cloud-provider/, Add `cidr-cattle-system: ${VIP_IP}/32` to kubevip configMap
```
Comment (jtravee) [inline docs/dev/dev-mode.md:107]:
```suggestion
   ## Refer to https://kube-vip.chipzoller.dev/docs/usage/cloud-provider/. Add `cidr-cattle-system: ${VIP_IP}/32` to kubevip configMap.
```

File: docs/dev/dev-mode.md [hunk 4]
```diff
@@ -13,70 +13,103 @@ Description: Developer mode (dev mode) is intended to be used for testing and de
...
+   kubectl -n kube-system edit cm kubevip
+
+   ## Change the rancher service type from ClusterIP to LoadBalancer, and then you can access Harvester UI via https://${VIP_IP}
```
Comment (jtravee) [inline docs/dev/dev-mode.md:110]:
```suggestion
   ## Change the rancher service type from ClusterIP to LoadBalancer, and then you can access Harvester UI via https://${VIP_IP}.
```

File: docs/dev/dev-mode.md [hunk 5]
```diff
@@ -13,70 +13,103 @@ Description: Developer mode (dev mode) is intended to be used for testing and de
...
+   helm install harvester-crd ./harvester-crd --namespace harvester-system
+   ```
 
-   ## Install the chart to the target namespace
-   helm install harvester harvester \
-   --namespace harvester-system \
-   --set longhorn.enabled=true,minio.persistence.storageClass=longhorn
+1. Install the Harvester chart
+   ```bash
+   ## in order to use the service type LoadBalancer and create a vip in control-plane nodes, we need to enable kubevip
```
Comment (jtravee) [inline docs/dev/dev-mode.md:86]:
```suggestion
   ## In order to use the service type LoadBalancer and create a vip in control-plane nodes, we need to enable kubevip.
```

Comment (jtravee) [reply docs/dev/dev-mode.md:86]: Just a suggestion to make the comments following the "##" start with capitals and end with punctuation for consistency.

File: docs/dev/dev-mode.md (hunk 1 above)
Comment (nunix) [reply docs/dev/dev-mode.md:26]: corrected, thank you

**Review by nunix — COMMENTED** [review]
Applied suggestions

**Review by jtravee — APPROVED** [review]

File: docs/dev/dev-mode.md (hunk 1 above)
Comment (futuretea) [reply docs/dev/dev-mode.md:26]: The controller-manager configuration is generally correct unless it has been specifically modified

**Review by nunix — APPROVED** [review]

**Review by btat — APPROVED** [review]

File: docs/dev/dev-mode.md [hunk 6]
```diff
@@ -13,70 +13,100 @@ Description: Developer mode (dev mode) is intended to be used for testing and de
 
 Developer mode (dev mode) is intended to be used for testing and development purposes.
```
Comment (guangbochen) [inline docs/dev/dev-mode.md:14]: Please help to add this one as an [attention text](https://squidfunk.github.io/mkdocs-material/reference/admonitions/#supported-types):
```
!!! attention
    Developer mode (dev mode) is intended to be used for local testing and development purposes.
```

Comment (futuretea) [reply docs/dev/dev-mode.md]: updated

File: docs/dev/dev-mode.md [hunk 7]
```diff
@@ -11,72 +11,103 @@ Description: Developer mode (dev mode) is intended to be used for testing and de
...
-    ```
-- If you are using an [RKE](https://rancher.com/docs/rke/latest/en/) cluster, `ipv4.ip_forward` must be enabled for the CNI plugin in order for pod network to work as expected.
 
 ## Installation
 
-Harvester can be installed on a Kubernetes cluster in the following ways:
-
-- Using the [Helm](https://helm.sh/) CLI.
-- As a Rancher catalog app, in which case the [harvester/harvester](https://github.com/harvester/harvester) repo is added to the Rancher Catalog as a Helm `v3` app
+Harvester can be installed on a Kubernetes cluster by using the [Helm](https://helm.sh/) CLI.
```
Comment (yasker) [inline docs/dev/dev-mode.md:29]: `For development purpose, Harvester can be installed...`

File: docs/dev/dev-mode.md [hunk 8]
```diff
@@ -11,72 +11,103 @@ Description: Developer mode (dev mode) is intended to be used for testing and de
...
 
 ## Requirements
 
+- The Kubernetes node must pass the [host-check](https://raw.githubusercontent.com/harvester/harvester/master/hack/host-check.sh)
+- If the Kubelet's RootDir is not `/var/lib/kubelet`, you must create a `bind mount` to `/var/lib/kubelet` as follows:
+   ```bash
+   KUBELET_ROOT_DIR="path to your kubelet root dir"
+   echo "${KUBELET_ROOT_DIR} /var/lib/kubelet none bind 0 0" >> /etc/fstab
+   mkdir -p /var/lib/kubelet && mount -a
 - [**Multus**](https://kubernetes.io/docs/concepts/cluster-administration/networking/#multus-a-multi-network-plugin) is installed across your cluster and a corresponding `NetworkAttachmentDefinition` CRD is created.
```
Comment (yasker) [inline docs/dev/dev-mode.md:25]: Do we require Longhorn and KubeVirt to be installed as well? Or it's a part of the chart.

**Review by yasker — CHANGES_REQUESTED** [review]

File: docs/dev/dev-mode.md (hunk 8 above)
Comment (futuretea) [reply docs/dev/dev-mode.md:25]: Harvester chart contains Longhorn and KubeVirt

File: docs/dev/dev-mode.md (hunk 7 above)
Comment (futuretea) [reply docs/dev/dev-mode.md]: updated

File: docs/dev/dev-mode.md (hunk 8 above)
Comment (yasker) [reply docs/dev/dev-mode.md:25]: Can we call that out here as well?

**Comment by yasker** [comment]
LGTM except for a minor comment on bundled Longhorn/KubeVirt.

File: docs/dev/dev-mode.md (hunk 8 above)
Comment (futuretea) [reply docs/dev/dev-mode.md:25]: Added

**Pull request merged**